lib/widgets/card.dart

```diff
@@ -0,0 +1,139 @@
+              Container(
+                margin: Constants.propertyContainerMargin,
+                decoration: BoxDecoration(
+                  borderRadius:
+                      BorderRadius.circular(Constants.containerRadius),
+                  color: Palette.primaryColor,
+                  boxShadow: Constants.containerShadow,
+                ),
+                child: Padding(
+                  padding: Constants.propertyContainerPadding,
+                  child: Column(
+                    children: [
+                      CardProperty(
+                        propertyName: Constants.cardIdLabel,
+                        text: card.cardId,
+                      ),
+                      CardProperty(
+                        propertyName: Constants.cardTypeLabel,
+                        text: card.type,
+                      ),
+                      CardProperty(
+                        propertyName: Constants.cardSetLabel,
+                        text: card.cardSet,
+                      ),
+                      CardProperty(
+                        propertyName: Constants.cardPlayerClassLabel,
+                        text: card.playerClass,
+                      ),
+                      CardProperty(
+                        propertyName: Constants.cardHealthLabel,
+                        text: card.health.toString(),
+                        visibility: card.health != null,
+                      ),
+                      CardProperty(
+                        propertyName: Constants.cardCostLabel,
+                        text: card.cost.toString(),
+                        visibility: card.cost != null,
+                      ),
+                      CardProperty(
+                        propertyName: Constants.cardAttackLabel,
+                        text: card.attack.toString(),
+                        visibility: card.attack != null,
+                      ),
+                      CardProperty(
+                        propertyName: Constants.cardMechanicsLabel,
+                        text: card.mechanics.toString(),
+                        visibility: card.mechanics != null,
+                      ),
+                    ],
+                  ),
+                ),
+              ),
+              Visibility(
+                visible: card.text != null,
+                child: Padding(
+                  padding: Constants.abilityLabelPadding,
+                  child: Column(
+                    mainAxisAlignment: MainAxisAlignment.center,
+                    children: [
+                      const Text(
+                        Constants.cardAbilityLabel,
+                        style: TextStyle(
+                          fontSize: Constants.abilityLabelFontSize,
+                          fontWeight: FontWeight.bold,
+                          color: Colors.black87,
+                        ),
+                      ),
+                      Text(
+                        card.text ?? "",
```

Comment:
You can add "" as emptyString to constant file

Reply:
Added